torch/lib/THPP/Tensor.hpp

```diff
@@ -36,6 +36,8 @@ struct Tensor {
   virtual bool isContiguous() const = 0;
   virtual void* data() = 0;
   virtual const void* data() const = 0;
+  virtual void* cdata() = 0;
+  virtual const void* cdata() const = 0;
```

Comment:
Is it really needed? The whole point of this class is to abstract the tensor type, and you'll have to cast this pointer to a concrete class afterwards. Why not `dynamic_cast<THTensor*>(t)->cdata()` (i.e. implement `cdata` only on concrete classes)?